src/components/AdminSessions/AdminSessions.js

```diff
@@ -0,0 +1,217 @@
+  state = {
+    sessions: [],
+    snackMessage: '',
+    openSnack: false,
+    open: false,
+    cinema: '',
+    films: [],
+    film: '',
+    numberOfSeats: 50,
+    time: new Date().toISOString().slice(0, 16)
+  };
+
+  getFilmsNames = () => {
+    axios
+      .get('http://localhost:3000/api/films')
+      .then(res => {
+        this.state.films = res.data.result;
+        for (const key of this.state.films) {
+          filmNames.push({ name: key.name, id: key._id });
+        }
+      })
+      .catch(() => {
+        this.setState({
+          errorMessage: 'Ошибка сервера'
+        });
+      });
+  };
+
+  handleClose = () => {
+    this.setState({ open: false });
+  };
+
+  handleClickOpen = () => {
+    this.setState({ open: true });
+  };
+
+  handleChange = name => event => {
+    this.setState({ [name]: event.target.value });
+  };
+
+  addSession = () => {
+    const { film, cinema, time } = this.state;
+    const emptySeats = this.state.numberOfSeats;
+    let filmId = 0;
+    for (const key of filmNames) {
+      if (key.name === film) {
+        filmId = key.id;
+        break;
+      }
+    }
+
+    axios
+      .post(
+        `http://localhost:3000/api/admin/sessions`,
+        { time, cinema, emptySeats, filmId },
+        {
+          headers: {
+            Authorization: 'JWT ' + UserService.getToken(),
+            'Content-Type': 'application/json'
+          }
+        }
+      )
+      .then(() => {
+        this.snack('Сеанс добавлен');
+        this.setState({ open: false });
+        this.getSessions();
+      })
+      .catch(err => {
+        this.snack('Не удалось добавить');
+        this.setState({ open: false });
+      });
+  };
+
+  getSessions = () => {
+    axios
+      .get('http://localhost:3000/api/admin/sessions', {
+        headers: {
+          Authorization: 'JWT ' + UserService.getToken(),
+          'Content-Type': 'application/json'
+        }
+      })
+      .then(res => {
+        this.setState({
+          sessions: res.data.sessions
+        });
+      })
+      .catch(() => {
+        this.setState({
+          errorMessage: 'Ошибка сервера'
+        });
+      });
+  };
+  componentDidMount() {
+    this.getSessions();
+    this.getFilmsNames();
+  }
+
+  snack = message => {
+    this.setState({ snackMessage: message, openSnack: true });
+    setTimeout(
+      () =>
+        this.setState({
+          openSnack: false
+        }),
+      3000
+    );
+  };
+
+  render() {
+    const { openSnack } = this.state;
+    return (
+      <div className="adminSessions__container">
+        <div className="sessionsList">
+          <Fab
+            color="primary"
+            className="addSession__button"
+            onClick={this.handleClickOpen}
+          >
+            <AddIcon />
+          </Fab>
+          {this.state.sessions.map(item => (
+            <AdminSessionCard
+              item={item}
+              snackbar={this.snack}
+              getSessions={this.getSessions}
```

Comment:
here and in other similar places:
do not pass functions to child components like this
create event in child component and handle it in a container
https://reactjs.org/docs/lifting-state-up.html